### ISSUE TITLE
WebSocket connection to 'wss://public-api.wordpress.com/pinghub/wpcom/me/newest-note-data' failed: Error during WebSocket handshake: Unexpected response code: 403

### ISSUE THREAD

**Issue by Jany-M** [Automattic/jetpack#9170]
I have been seeing this error on several sites/server configs in the past months:

`WebSocket connection to 'wss://public-api.wordpress.com/pinghub/wpcom/me/newest-note-data' failed: Error during WebSocket handshake: Unexpected response code: 403`

It was also raised [here](https://github.com/Automattic/pinghub/issues/8) and has been [asked](https://wordpress.org/support/topic/error-after-installing-jetpack-2/) in _wordpress.org_ forums as well as _wordpress.com_ [ones](https://en.forums.wordpress.com/topic/help-with-error-during-websocket-handshake-unexpected-response-code-403).
None of the answers fix the problem nor really fully address it.

Besides being logged into _wordpress.com_, is there a real solution to getting rid of this error in the console?

**Comment by designsimply** [comment]
Noting a few additional reports:

* https://wordpress.org/support/topic/websocket-connection-to-wsspublic-apiwordpresscompinghubwp-failed-403/
* https://stackoverflow.com/questions/36900447/websocket-connection-failure-error-during-websocket-handshake-response-code-40

**Comment by jeherve** [comment]
Thanks for the report. This is currently being tracked internally, in 604-gh-notifications-client.

Until the issue is resolved, I would recommend one of the following work-arounds:

- Log in to your WordPress.com account [here](https://wordpress.com/log-in).
- If you use the Jetpack plugin on your site, go to this page and deactivate the Notifications feature:
`https://yoursite.com/wp-admin/admin.php?page=jetpack_modules`

Closing as duplicate of 604-gh-notifications-client.

**Comment by ChrisWiegman** [comment]
Just to add here that logging into WordPress.com doesn't fix the issue. I am logged in in Chrome and am still seeing the issue on my Jetpack-connected site.

**Comment by kraftbj** [comment]
@ChrisWiegman Could you send any header/response information from the browser console to us via https://jetpack.com/contact-support/ ? I haven't seen a case personally where logging into WordPress.com didn't resolve the issue.

**Comment by davidjbrossard** [comment]
Has this been solved? I am seeing the same error on a site I manage (delscookingtwist.com). Thanks.

**Comment by kraftbj** [comment]
@davidjbrossard Can you verify that you're logged into WordPress.com in that same browser? The error is expected when there is not an active WordPress.com session.

**Comment by davidjbrossard** [comment]
On a clean browser, having logged into wordpress.com first, I no longer see the error. Thanks.

**Comment by kraftbj** [comment]
@unknys are you saying that logging into WordPress.com doesn't work or that you want a different solution?

Is there a functional issue you're seeing from these?

At this time, the 403 notice is expected when you're not logged into WordPress.com and while we're tracking this issue, we do not have a timeline for when we'll look at addressing it.

**Comment by unknys** [comment]
@krafrbj

That is exactly what i am saying. I also have done the following since everyone on support for wordpress.com/ord thinks its a jetpack problem.

But i have disabled the jetpack. And the problem still occur.

It seems it tries to connect to wordpress and wp.me.

So if jetpack is disabled and the same error comes up i doubt this has anything to do with jetpack. If you read the error messages. Really read them. And they are copied in here after jetpack was disabled.

Someone told me it was a http/1 problem and that it worked when they changed to using https/2.

**Comment by jeherve** [comment]
@unknys Could you contact us via [this form](https://jetpack.com/contact-support/?rel=support) and give us more details about your site there, so we can take a closer look at the issue with you?

Thank you!

**Comment by joshchretien** [comment]
Have you found a fix to this? I get the same error

**Comment by jeherve** [comment]
@joshchretien Could you try the work-around here until this gets fixed?
https://github.com/Automattic/jetpack/issues/9170#issuecomment-376867824

**Comment by sparanoid** [comment]
I also got this error when visiting https://wordpress.com/stats/day:

```
WebSocket connection to 'wss://public-api.wordpress.com/pinghub/wpcom/me/newest-note-data' failed: Error during WebSocket handshake: Unexpected response code: 403
```

It looks like this connection is blocked by wordpress.com backend.

**Comment by zirtrex** [comment]
I have the same error, please notificate for any solution.

**Comment by JohnRDOrazio** [comment]
This issue keeps popping up with new Wordpress updates. I just updated to the latest 5.3.2 and starting getting this error yet again. Am logged in to Wordpress.com on the same browser but error still there:

WebSocket connection to 'wss://public-api.wordpress.com/pinghub/wpcom/me/newest-note-data' failed: Error during WebSocket handshake: Unexpected response code: 403
v @ [?v=2.0:6](https://public-api.wordpress.com/wp-admin/rest-proxy/?v=2.0)
T @ [?v=2.0:6](https://public-api.wordpress.com/wp-admin/rest-proxy/?v=2.0)
o @[ ?v=2.0:6](https://public-api.wordpress.com/wp-admin/rest-proxy/?v=2.0)

The resource referred to is https://public-api.wordpress.com/wp-admin/rest-proxy/?v=2.0 .
If I inspect the network requests I can see the request headers:

Request URL: wss://public-api.wordpress.com/pinghub/wpcom/me/newest-note-data
Provisional headers are shown
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.9,it;q=0.8
Cache-Control: no-cache
Connection: Upgrade
Host: public-api.wordpress.com
Origin: https://public-api.wordpress.com
Pragma: no-cache
Sec-WebSocket-Extensions: permessage-deflate; client_max_window_bits
Sec-WebSocket-Key: YUfdwsONaJcvTkLvsj9+GA==
Sec-WebSocket-Version: 13
Upgrade: websocket
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.149 Safari/537.36

**Comment by jeherve** [comment]
> If I remember correctly when the error popped up a couple years ago I had to completely wipe Jetpack from my site and the do fresh re-install, and only then did this error go away.

That most likely won't help, as the error happens on WordPress.com. You can rest assured though, this won't impact any of the Jetpack features on your site, it only happens within the Notifications panel that is added to your site's admin bar and does not impact any other features.

> I have also noticed that this seems to be creating many many transients, I have about 150 transients that look like this

Those transients are created by the Protect feature, so they're unrelated to the error you're seeing. They should get deleted by WordPress fairly quickly though, as they have a short expiration date. If expired transients are not being deleted from your site, I would recommend looking into any plugins that may change the behaviour of transients. It could be a caching plugin or a database optimization plugin. If transients are deleted properly, you shouldn't run into any issues.

I hope this clarifies things a bit.

**Comment by JohnRDOrazio** [comment]
I can confirm that disabling the notifications module does prevent the error from showing. There doesn't seem to be any way of getting there through the admin interface, but the direct link that you gave "/wp-admin/admin.php?page=jetpack_modules" does work

**Comment by andreaguerra80** [comment]
Hi,
I also have this error, over 30 notifications in my google console, all the same.

WebSocket connection to 'wss://public-api.wordpress.com/pinghub/wpcom/me/newest-note-data' failed: Error during WebSocket handshake: Unexpected response code: 403 ?v=2.0:6

**Comment by jeherve** [comment]
> There doesn't seem to be any way of getting there through the admin interface

@JohnRDOrazio You can get to that page by clicking on "Modules" in the footer of the Jetpack dashboard.

**Comment by csonnek** [comment]
Another report of this in 2971386-zen - advising they disable notifications

**Comment by kraftbj** [comment]
Generally, this is still when someone is not logged into WordPress.com. First response for someone asking about it is checking if they're logged in.

**Comment by supernovia** [comment]
@kraftbj we've got a case here where someone is logged in but still gets the error. Possibly due to Safari's new cookie rules?
https://wordpress.org/support/topic/403-forbiden-public-api-wordpress-com-pinghub-wpcom-me-newest-note-data-2/

**Comment by kraftbj** [comment]
@supernovia That's definitely possible, especially on Safari.

**Comment by eliasasaid1** [comment]
hello any solution for this?

**Comment by brown7jazz** [comment]
Hi.
I had exactly the same error on a medical hours booking website. It looped me and generated duplicate appointments, unpaid appointments did not remove from the calendar, everything was a disaster.
Disable the plugins:
- JetPack (I don't use it)
- Autoptimize (Cache Plugin)

If they're not using auto-optimization, try disabling the caching plugin for the site you're using.

I hope it serves you.

Bye.

**Comment by tmmbecker** [comment]
3517667-zen - advised that they disable notifications.

**Comment by UberEclectic** [comment]
This is closed as dupe of 604-gh-notifications-client.  How do we track the progress of 604-gh-notifications-client?

**Comment by jeherve** [comment]
> How do we track the progress of 604-gh-notifications-client?

This is tracked internally, but we'll update this issue when we make progress on that problem.

**Comment by codywohlers** [comment]
I had this issue and I was already logged in, but logging out and then logging back in fixed it.

**Comment by psmits1567** [comment]
I did not have this problem before until today. Logging in and out does not help.
I do not use jetpack, it is the translate.wordpress.org site that causes it
WebSocket connection to 'wss://public-api.wordpress.com/pinghub/wpcom/me/newest-note-data' failed:

**Comment by jeherve** [comment]
> I do not use jetpack, it is the translate.wordpress.org site that causes it

That website uses the Jetpack plugin, hence why you're seeing the issue. You can safely ignore the error though; it will not impact your browsing experience or translating on translate.wordpress.org.

We're currently tracking this issue here:
https://github.com/Automattic/wp-calypso/issues/34272

**Comment by psmits1567** [comment]
Well then something has changed, because I did not see it before. While investigating this issue I noticed that if you signon to WordPress.com the error is gone. I tested that and indeed after signon to WordPress.com the error is gone. It seems that translate.wordpress.org is currently using the messaging system from WordPress.com. because after signon I got messages from WordPress.com. Although translate.wordpress.org is using JetPack, it should not give this warning if you do not signon to WordPress.com. As most of us do not have a WordPress.com account.
Thanks for your response, and I hope there will be a solution soon

**Comment by neptunoec** [comment]
2022  and the issue persist

**Comment by Steveorevo** [comment]
Also experiencing this issue. Turning off notifications as suggested by jeherve, or removing Jetpack resolves the issue.

_If you use the Jetpack plugin on your site, go to this page and deactivate the Notifications feature:
https://yoursite.com/wp-admin/admin.php?page=jetpack_modules_

**Comment by paxperscientiam** [comment]
Just wanna echo that disabling the notifications module did it. Thanks @Steveorevo

**Comment by colonelclick** [comment]
That did not fix it for me, but deactivating Jetpack entirely did the trick.

**Comment by therealestpetie** [comment]
Miine is saying wp-content/plugins/all-in-one-wp-migration/lib/view/assets/javascript/import.min.js?v=7.62&ver=6.0.1:1345

I have tried almost everything

**Comment by therealestpetie** [comment]
error: function error(jqXHR, textStatus) {
      throw new Error(textStatus);

**Comment by majick777** [comment]
Found this for anyone wanting to do this programmatically on a number of sites:
https://blog.room34.com/archives/6482/get-rid-of-that-annoying-websocket-connection-to-failed-error-on-wordpress-sites-running-jetpack/

I was getting this on the frontend also when logged in so modified that function to this:
```
add_action( 'init', 'jetpack_deactivate_notes_module' );
function jetpack_deactivate_notes_module() {
    if ( !is_user_logged_in() ) {return;}
    if ( class_exists( 'Jetpack' ) && Jetpack::is_module_active( 'notes' ) ) {
        Jetpack::deactivate_module( 'notes' );
    }
}
```

**Comment by GeoBobby** [comment]
Same problem here: https://freespinsnodeposit.net/free-spins-no-deposit-new-zealand/ 
![Screenshot from 2023-01-17 19-15-13](https://user-images.githubusercontent.com/122912968/212967212-e1416b51-4518-4755-8e60-3efbc9bc7b07.png)

How can we solve this issue? This error have an impact on website functionality?

**Comment by jeherve** [comment]
> This error have an impact on website functionality?

No, it doesn't, you can rest assured.

> How can we solve this issue?

While this should not impact your user experience at all, if you want to see the error gone, I would recommend that you try the following work-arounds suggested here:
https://github.com/Automattic/jetpack/issues/9170#issuecomment-376867824

**Comment by GeoBobby** [comment]
> > This error have an impact on website functionality?
> 
> No, it doesn't, you can rest assured.
> 
> > How can we solve this issue?
> 
> While this should not impact your user experience at all, if you want to see the error gone, I would recommend that you try the following work-arounds suggested here: [#9170 (comment)](https://github.com/Automattic/jetpack/issues/9170#issuecomment-376867824)

Thanks man! 
Wish you all the best!

**Comment by jgarcianewemage** [comment]
FIXED: uninstall jetpack, it's malware

**Comment by zohebnsr** [comment]
After all these years from 2018 to 2024...... The error still exists. 6 Years of failure? And the top answers here is to remove Jetpack plugin to fix the issue. What a shame!

**Comment by H4wKs** [comment]
> After all these years from 2018 to 2024...... The error still exists. 6 Years of failure? And the top answers here is to remove Jetpack plugin to fix the issue. What a shame!

I was investigating this issue for a friend, and at least your comment made my day @zohebnsr 

In my case, this [comment](https://github.com/Automattic/jetpack/issues/9170#issuecomment-376867824) did the trick which consist of deactivating the notifications inside JetPack module.

But I agree, crazy that no one care enough fixing this for the past 6 years

**Comment by php4fan** [comment]
Why the f*** is this closed??

**Comment by jeherve** [comment]
> Why the f*** is this closed??

You can find out more in my comment here, back when I closed the issue:
https://github.com/Automattic/jetpack/issues/9170#issuecomment-376867824

I will now lock the conversation on this issue since we've discussed about the impact and workarounds above, and since the issue is tracked elsewhere already. If you still experience problems on your own site:

1. Log in to your WordPress.com account [here](https://wordpress.com/log-in). This has helped the folks above on this issue.
2. If that doesn't help and if you use the Jetpack plugin on your site, deactivate the Notifications feature [as explained here](https://jetpack.com/support/control-jetpack-features-on-one-page/).
3. If you'd like to comment about the problem, you can comment on this issue: https://github.com/Automattic/wp-calypso/issues/34272
